fix: examples build produces 20 warnings (CA1050, CA1305)

## Problem

`bash scripts/build-local.sh` step 5 ("Build examples") succeeds but produces **20 warnings**. We require a fully green build with zero warnings.

```
Examples net8.0 succeeded with 20 warning(s)
```

### CA1050 — "Declare types in namespaces" (19 warnings)

All example files under `examples/` declare classes in the global namespace. Affected files:

- `Admin.cs`, `Authorization.cs`, `BatchOperation.cs`, `Client.cs`, `Decision.cs`, `Deployment.cs`, `Document.cs`, `Group.cs`, `Incident.cs`, `Job.cs`, `MappingRule.cs`, `MessageSignal.cs`, `ProcessDefinition.cs`, `ProcessInstance.cs`, `Role.cs`, `Tenant.cs`, `User.cs`, `UserTask.cs`, `VariableElement.cs`

Example:
```
examples/Deployment.cs(5,21): warning CA1050: Declare types in namespaces
```

### CA1305 — "Provide a value for the IFormatProvider argument" (1 warning)

```
docs/examples/ReadmeExamples.cs(451,22): warning CA1305: The behavior of 'LoggerSinkConfiguration.Console(...)' could vary based on the current user's locale settings.
```

## Suggested fix

**CA1050**: Either wrap each example class in a namespace (e.g. `namespace Camunda.Orchestration.Sdk.Examples;`) or suppress the rule for the examples project via `.editorconfig` / `<NoWarn>` in the examples `.csproj`.

**CA1305**: Pass `CultureInfo.InvariantCulture` (or similar) to the Serilog `.Console()` call, or suppress if not applicable.

## Acceptance criteria

- `bash scripts/build-local.sh` produces **0 warnings** in step 5 (build examples)

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -5,9 +5,8 @@
     <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
     <AnalysisLevel>latest-recommended</AnalysisLevel>
 
-    <!-- Treat all analyzer warnings as warnings (not errors) during build.
-         CI will gate separately via dotnet format with verify-no-changes. -->
-    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
+    <!-- Make all warnings fatal so builds fail on any warning (local + CI). -->
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
   </PropertyGroup>
 
 </Project>

--- a/docs/examples/Examples.csproj
+++ b/docs/examples/Examples.csproj
@@ -9,6 +9,8 @@
     <OutputType>Library</OutputType>
     <NoDefaultLaunchSettingsFile>true</NoDefaultLaunchSettingsFile>
     <RootNamespace>Camunda.Orchestration.Sdk.Examples</RootNamespace>
+    <!-- Suppress analyser rules that aren't meaningful for compile-only example code -->
+    <NoWarn>CA1050;CA1305</NoWarn>
   </PropertyGroup>
 
   <ItemGroup>

